migrate/starter/starter.go

```diff
@@ -79,42 +97,55 @@ func checkStartVersion(args []string) version {
 	switch walVersion {
 	case wal.WALv0_5:
 		return internalV2
+	case wal.WALv0_5Proxy:
+		return internalV2Proxy
 	case wal.WALv0_4:
-		// TODO: standby case
-		// if it is standby guy:
-		//     print out detect standby mode
-		//     go to WALNotExist case
-		//     if want to start with 2.0:
-		//         remove old data dir to avoid auto migration
-		//         try to let it fallback? or use local proxy file?
-		ver, err := checkStartVersionByDataDir4(dataDir)
-		if err != nil {
-			log.Fatalf("migrate: failed to check start version in %v: %v", dataDir, err)
+		standbyInfo, err := migrate.DecodeStandbyInfo4FromFile(standbyInfo4(dataDir))
+		switch {
+		case os.IsNotExist(err):
```

Comment:
is this correct? 
if version == internalV2 but mode in standby is etcdNode, should'nt start as etcd node rather than v2 proxy?